next.config.js

```diff
@@ -1,22 +1,123 @@
+                maxAgeSeconds: 365 * 24 * 60 * 60, // 365 days
+            },
+        },
+    },
+    {
+        urlPattern: /\.(?:eot|otf|ttc|ttf|woff|woff2|font.css)$/i,
+        handler: 'StaleWhileRevalidate',
+        options: {
+            cacheName: 'static-font-assets',
+            expiration: {
+                maxEntries: 4,
+                maxAgeSeconds: 7 * 24 * 60 * 60, // 7 days
+            },
+        },
+    },
+    {
+        urlPattern: /\.(?:jpg|jpeg|gif|png|svg|ico|webp)$/i,
+        handler: 'StaleWhileRevalidate',
+        options: {
+            cacheName: 'static-image-assets',
+            expiration: {
+                maxEntries: 64,
+                maxAgeSeconds: 24 * 60 * 60, // 24 hours
+            },
+        },
+    },
+    {
+        urlPattern: /\.(?:js)$/i,
+        handler: 'StaleWhileRevalidate',
+        options: {
+            cacheName: 'static-js-assets',
+            expiration: {
+                maxEntries: 32,
+                maxAgeSeconds: 24 * 60 * 60, // 24 hours
+            },
+        },
+    },
+    {
+        urlPattern: /\.(?:css|less)$/i,
+        handler: 'StaleWhileRevalidate',
+        options: {
+            cacheName: 'static-style-assets',
+            expiration: {
+                maxEntries: 32,
+                maxAgeSeconds: 24 * 60 * 60, // 24 hours
+            },
+        },
+    },
+    {
+        urlPattern: /\.(?:json|xml|csv)$/i,
+        handler: 'NetworkFirst',
+        options: {
+            cacheName: 'static-data-assets',
+            expiration: {
+                maxEntries: 32,
+                maxAgeSeconds: 24 * 60 * 60, // 24 hours
+            },
+        },
+    },
+    {
+        urlPattern: /\/api\/.*$/i,
+        handler: 'NetworkFirst',
+        method: 'GET',
+        options: {
+            cacheName: 'apis',
+            expiration: {
+                maxEntries: 16,
+                maxAgeSeconds: 24 * 60 * 60, // 24 hours
+            },
+            networkTimeoutSeconds: 10, // fall back to cache if api does not response within 10 seconds
+        },
+    },
+];
 
 module.exports = withPWA({
     pwa: {
         dest: 'public',
-        disable: process.env.NODE_ENV !== 'production'
+        disable: process.env.NODE_ENV !== 'production',
+        runtimeCaching: pwaCacheOptions,
     },
-    webpack: config => {
+    webpack: (config) => {
         config.module.rules.unshift({
             test: /\.worker\.(js|ts)$/,
             loader: 'worker-loader',
             options: {
                 name: 'static/[hash].worker.js',
-                publicPath: '/_next/'
-            }
-        })
+                publicPath: '/_next/',
+            },
+        });
 
-        config.output.globalObject = `(typeof self !== 'undefined' ? self : this)`
+        config.output.globalObject = `(typeof self !== 'undefined' ? self : this)`;
 
-        return config
-    }
-});
\ No newline at end of file
+        return config;
+    },
```

Comment:
What are these commas (line 116 and 122) for if there is nothing coming after them?